crates/nu-explore/src/views/record/mod.rs

```diff
@@ -23,25 +23,39 @@ use nu_protocol::{
     Config, Record, Span, Value,
 };
 use ratatui::{layout::Rect, widgets::Block};
-use std::{borrow::Cow, collections::HashMap};
+use std::collections::HashMap;
 
 pub use self::table_widget::Orientation;
 
 #[derive(Debug, Clone)]
-pub struct RecordView<'a> {
-    layer_stack: Vec<RecordLayer<'a>>,
+pub struct RecordView {
+    layer_stack: Vec<LayerData>,
     mode: UIMode,
     orientation: Orientation,
     cfg: ExploreConfig,
 }
 
-impl<'a> RecordView<'a> {
-    pub fn new(
-        columns: impl Into<Cow<'a, [String]>>,
-        records: impl Into<Cow<'a, [Vec<Value>]>>,
-    ) -> Self {
+#[derive(Debug, Clone)]
+struct LayerData {
+    record: RecordLayer,
+    widget_data: Option<RecordData>,
+}
+
+#[derive(Debug, Clone)]
+struct RecordData {
+    columns: Vec<String>,
```

Comment:
Do we need to store columns in `RecordData` when `RecordLayer` also contains columns?

Reply:
Basically no;